addons/hr_holidays/models/hr_leave.py

```diff
@@ -754,7 +754,7 @@ def _check_approval_update(self, state):
         if self.env.user.id == SUPERUSER_ID:
             return
 
-        current_employee = self.env['hr.employee'].search([('user_id', '=', self.env.uid)], limit=1)
+        current_employee_id = self.env['hr.employee'].search([('user_id', '=', self.env.uid)], limit=1).id
```

Comment:
Do we need to do this? `self.env['hr.employee'].search()` should return an employee because only `_search()` is overriden, so it should make no difference.
If it does, we need a big warning comment here - this business with comparing IDs only is too mysterious.